src/main/java/org/mvplugins/multiverse/core/teleportation/AsyncSafetyTeleporter.java

```diff
@@ -0,0 +1,164 @@
+package org.mvplugins.multiverse.core.teleportation;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import com.dumptruckman.minecraft.util.Logging;
+import io.papermc.lib.PaperLib;
+import io.vavr.control.Try;
+import jakarta.inject.Inject;
+import org.bukkit.Location;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.Player;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jvnet.hk2.annotations.Service;
+
+import org.mvplugins.multiverse.core.api.BlockSafety;
+import org.mvplugins.multiverse.core.destination.ParsedDestination;
+import org.mvplugins.multiverse.core.utils.result.Result;
+
+@SuppressWarnings("unchecked")
+@Service
+public class AsyncSafetyTeleporter {
```

Comment:
Is it worth distinguishing this as Async if we don't have an alternative?

Reply:
Good point